java/src/main/java/com/logicalclocks/hsfs/StreamFeatureGroup.java

```diff
@@ -428,4 +429,18 @@ public String getEncodedAvroSchema() throws FeatureStoreException, IOException {
   public Schema getDeserializedAvroSchema() throws FeatureStoreException, IOException {
     return utils.getDeserializedAvroSchema(getAvroSchema());
   }
+
+  public List<String> nonNullableColumns() {
```

Comment:
Can't we have this in the feature group base, so we don't have to have it twice?

Reply:
We don't have the required properties there unfortunately.

Reply:
ah right, partition and pre-combine.